### PR TITLE
PS-5794: Crash when innodb-redo-log-encrypt=ON specified during boots…

### PR DESCRIPTION
**Pull request by dutow**
…trap

### CODE REVIEW

File: mysql-test/suite/innodb/t/percona_bug_ps5794.test [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# restart
```
Comment (laurynas-biveinis) [inline mysql-test/suite/innodb/t/percona_bug_ps5794.test]: Remove

File: mysql-test/suite/innodb/t/percona_bug_ps5794.test [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+--echo # Stop the MTR default DB server
+--source include/shutdown_mysqld.inc
+
+--let $KEYRING_PARAMS=--early-plugin-load="keyring_file=$KEYRING_PLUGIN" --loose-keyring_file_data=$MYSQL_TMP_DIR/mysecret_keyring $KEYRING_PLUGIN_OPT
+let extra_args=--no-defaults --console --loose-skip-auto_generate_certs --loose-skip-sha256_password_auto_generate_rsa_keys --skip-ssl --basedir=$BASEDIR --lc-messages-dir=$MYSQL_SHAREDIR;
+let NEW_CMD = $MYSQLD $extra_args --innodb_dedicated_server=OFF --initialize-insecure $KEYRING_PARAMS --datadir=$MYSQLD_DATADIR1 --init-file=$BOOTSTRAP_SQL --secure-file-priv="" --innodb-redo-log-encrypt >>$MYSQL_TMP_DIR/bootstrap1.log 2>&1;
+
+--echo # Run the bootstrap command of datadir1
+--exec $NEW_CMD
+--sleep 5
```
Comment (laurynas-biveinis) [inline mysql-test/suite/innodb/t/percona_bug_ps5794.test]: Please sync properly

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
5.7?

**Comment by satya-bodapati** [comment]
This looks like wrong fix. Enum shouldn't allow optional arguments. Option value is must.

for
  static MYSQL_SYSVAR_ENUM(redo_log_encrypt, srv_redo_log_encrypt,

you should use PLUGIN_VAR_RQCMDARG instead of PLUGIN_VAR_OPCMDARG

**Comment by dutow** [comment]
Upstream allows it to be empty - so we should allow it to be empty. Also, even if I disallow the empty version, the issue would be there with `=ON`, as that's basically the empty option.

**Comment by dutow** [comment]
In 5.7,  we don't allow the "ON" option, it's not affected.

**Review by satya-bodapati — APPROVED** [review]
Please move this testcase to existing redo log encrypt testcase

**Comment by dutow** [comment]
@satya-bodapati I kept the test where it is - other test either have a specific topic or have mk/rk variants both.

**Review by laurynas-biveinis — APPROVED** [review]

**Comment by dutow** [comment]
https://ps80.cd.percona.com/job/percona-server-8.0-param/274/

**Pull request merged**